### PR TITLE
Add carousel magazine missing alarm/stop.

### PR DESCRIPTION
**Pull request by StephanCoetzee**
These changes cause the carousel to stop when a magazine is missing. It uses the probe sensor. The name of the probe sensor is changed to magazine alignment sensor. However, all probe functionality of the original Grbl is kept.

The alarm can be cleared in a similar fashion to other alarms and the motors are de-energized when the alarm is issued, similar to other alarms.

The alarm is issued whenever the carousel moves and a magazine is missing, including when the carousel is homing.

### CODE REVIEW

File: counters.c
Comment (Jeff-Ciesielski) [inline counters.c]: Missing whitespace around `&` operator

File: cpu_map_keyme2560.h
Comment (Jeff-Ciesielski) [inline cpu_map_keyme2560.h]: Whitespace around `<<`

File: limits.c
Comment (Jeff-Ciesielski) [inline limits.c:239]: Operator whitespace (line 238 and 241)

File: magazine.c
Comment (Jeff-Ciesielski) [inline magazine.c]: Inconsistent brace style in this file.

Seems like a good portion of the grbl codebase follows the convention of the brace on the following line (which is of course, the one true way ;) ).

File: magazine.c
Comment (Jeff-Ciesielski) [inline magazine.c]: Whitespace around `*`

File: magazine.c
Comment (Jeff-Ciesielski) [inline magazine.c]: This should be const.

File: magazine.c
Comment (Jeff-Ciesielski) [inline magazine.c]: This line is too long (should always try to be < 80 characters).

Given that we're using a pretty aggressive optimization setting, there's no reason to not break this up (it'll get folded down to very similar machine code), especially if you give the compiler a hint about value aliasing via `const`.

Something like:

```
const int32_t sys_pos = sys.position[C_AXIS];
const int32_t probe_pos = sys.probe_position[C_AXIS];
const int32_t delta_pos = (abs(sys_pos - probe_pos) / 
                           settings.steps_per_mm[C_AXIS]);

if (delta_pos > settings.mag_gap_limit) { 
...
}
```

File: magazine.c
Comment (Jeff-Ciesielski) [inline magazine.c]: Missing space around `*`

File: magazine.c
Comment (Jeff-Ciesielski) [inline magazine.c:41]: (Not for this PR, but something to think about)

As we briefly discussed, might be worth it to wrap this little bit of logic up in a function or a macro so that all required flags get set.

File: probe.h
Comment (Jeff-Ciesielski) [inline probe.h]: Makes me nervous to leave commented out code laying around (and to have changed the values in this macro...).

Might be better to simply alias PROBE_PIN and PROBE_MASK to their magazine equivalents with another macro and leave them as PROBE_PIN/PROBE_MASK in this macro (this will also make it easier to adjust these values later should we re-purpose this functionality)

**Review by Jeff-Ciesielski — CHANGES_REQUESTED** [review]
Overall, I think the logic looks good, most of my nits here are just readability related.

Nice job.

File: limits.c
Comment (Jeff-Ciesielski) [inline limits.c]: Commented out code, pls remove.

File: limits.c
Comment (Jeff-Ciesielski) [inline limits.c]: This should probably be setting mag_gap_check equal to the MAG_GAP_CHECK_ENABLE define (so that this behavior is obeyed)

**Comment by mdicicco** [comment]
Clarification on the need for this: It will also detect missing magazine slats, but it's more important that we catch the case where the belt is stuck on something and not moving.  If we spin the carousel very far with something blocking it, chaos ensues.

I'll do full code review tomorrow, but I figured it was worth noting that for posterity.

File: cpu_map_keyme2560.h
Comment (mdicicco) [inline cpu_map_keyme2560.h:115]: much better name, thanks for this.

File: defaults.h
Comment (mdicicco) [inline defaults.h:112]: We might want to consider over-writing this via a EEPROM setting.  I could see this as something we might want to configure.  Just a though.

File: magazine.c
Comment (StephanCoetzee) [reply magazine.c]: Ok, will change.

File: magazine.c
Comment (StephanCoetzee) [reply magazine.c]: I understand the reasoning behind some of the stylistic suggestions regarding whitespace, but most of it was written to follow the same convention as the existing Grbl code. If I change it, it will look different than many other lines, often in the same file as my changes, especially if we want to make pull requests to the original Grbl later down the line.

File: magazine.c
Comment (StephanCoetzee) [reply magazine.c]: Ok

File: magazine.c
Comment (StephanCoetzee) [reply magazine.c]: Ok

File: magazine.c
Comment (StephanCoetzee) [reply magazine.c:41]: The problem with writing a function to set these types of flags is that the number of flags that gets set is somewhat specific to the situation. There are occasions where only a single flag is set, such as SYS_EXEC |= EXEC_FEED_HOLD. There are other occasions where  some of these flags are set, but sys.state does not change, etc.

File: limits.c
Comment (StephanCoetzee) [reply limits.c]: Ok, will remove.

File: limits.c
Comment (StephanCoetzee) [reply limits.c]: You're right, it should change.

File: defaults.h
Comment (StephanCoetzee) [reply defaults.h:112]: Makes sense. I'll change it.

File: magazine.c
Comment (Jeff-Ciesielski) [reply magazine.c]: If it were bracing style, capitalization, or indentation, I'd agree, but whitespace and parens around mathematical operators are typically the one place I think it always pays to be conservative and include them.  I understand your point about upstream inclusion, but:

1) We still have to be the active maintainers of this branch of the code as it is a core part of our product.  Increased readability in our code has no down side and can be very helpful in tracking down weird operator precedence and math bugs.
-and-
2) Upstream inclusion isn't guaranteed even if we did submit a PR, and they don't seem to have a clear indication of actual expected coding style  (via a contributing document or a how-to).  If there were complaints from the upstream maintainers, they would be easy to address (or we could discuss a codebase-wide refactor to help readability).

**Comment by Jeff-Ciesielski** [comment]
:+1:

File: probe.c
Comment (mdicicco) [inline probe.c]: What was the reason for adding all these includes if they don't get used in this file?

**Comment by mdicicco** [comment]
One minor issue, other than that I think this is good.  re-squash after you make any final edits and I can merge.

File: probe.c
Comment (StephanCoetzee) [reply probe.c]: Good point. I took them out and amended the commit. I added these headers after I made changes in the file. I then took out the changes but forgot about the headers.

**Pull request merged**
